src/replit/ai/modelfarm/google/language_models/text_generation_model.py

```diff
@@ -109,15 +124,17 @@ def __ready_response(
         Returns:
             TextGenerationResponse: The transformed response.
         """
-        choice = response.responses[0].choices[0]
-        safetyAttributes = choice.metadata["safetyAttributes"]
+        choice = response.choices[0]
+        safetyAttributes = choice.metadata[
+            "safetyAttributes"] if choice.metadata else {}
         safetyCategories = dict(
-            zip(safetyAttributes["categories"], safetyAttributes["scores"], strict=True)
-        )
+            zip(safetyAttributes["categories"],
```

Comment:
Could produce a KeyError if `safetyAttributes` is empty. Not sure if this is actually possible in reality though.